conans/client/graph/range_resolver.py

```diff
@@ -147,37 +145,50 @@ def _resolve_local(self, search_ref, version_range):
+                remote_results = self._remote_manager.search_recipes(remote, pattern, ignorecase=False)
+                remote_results = [ref for ref in remote_results
+                                  if ref.user == search_ref.user and ref.channel == search_ref.channel]
+                resolved_version = self._resolve_version(version_range, remote_results)
+                if resolved_version and not check_all_remotes:
+                    return resolved_version, remote.name
+                elif resolved_version:
+                    results.append({"remote": remote.name,
+                                    "version": resolved_version})
+        if len(results) > 0:
+            resolved_version = self._resolve_version(version_range,
+                                                     [result.get("version") for result in results])
+            for result in results:
+                if result.get("version") == resolved_version:
+                    return result.get("version"), result.get("remote")
+        else:
+            return None, None
+
+    def _resolve_remote(self, search_ref, version_range, remotes, check_all_remotes):
         # We should use ignorecase=False, we want the exact case!
         found_refs, remote_name = self._cached_remote_found.get(search_ref, (None, None))
-        if found_refs is None:
+        # we have to check that the resolved cached ref is in the range
+        if found_refs is None or not self._resolve_version(version_range, [found_refs]):
             # Searching for just the name is much faster in remotes like Artifactory
-            found_refs, remote_name = self._search_remotes(search_ref, remotes)
+            found_refs, remote_name = self._search_remotes(search_ref, version_range, remotes,
+                                                           check_all_remotes)
             if found_refs:
                 self._result.append("%s versions found in '%s' remote" % (search_ref, remote_name))
             else:
                 self._result.append("%s versions not found in remotes")
             # We don't want here to resolve the revision that should be done in the proxy
             # as any other regular flow
-            found_refs = [ref.copy_clear_rev() for ref in found_refs or []]
+            # FIXME: refactor all this and update for 2.0
+            found_refs = found_refs.copy_clear_rev() if found_refs else None
             # Empty list, just in case it returns None
             self._cached_remote_found[search_ref] = found_refs, remote_name
-        if found_refs:
-            return self._resolve_version(version_range, found_refs), remote_name
-        return None, None
+        return (found_refs, remote_name) if found_refs else (None, None)
 
     def _resolve_version(self, version_range, refs_found):
-        versions = {ref.version: ref for ref in refs_found}
+        versions = {ref.version: ref for ref in refs_found if ref}
```

Comment:
This ref is a bit concerning, can it be None? From where?

Reply:
I added that because of this addition: https://github.com/conan-io/conan/blob/0c0c2e28972ab9acd7f2add0ab21012a27c2c638/conans/client/graph/range_resolver.py#L125-L127
I thought that maybe if that code does not resolve a reference in the remotes this could fail